### DISCUSSION TITLE
Is page variable accesible in macros?

### DISCUSSION THREAD

**Discussion by Guirdo** [11ty/eleventy#2846] [Q&A]
I was working with some nunjucks macros and when I tried to use `eleventy-plugin-i18n` inside of them, it used the fallback because it seens that the [`page`](https://www.11ty.dev/docs/data-eleventy-supplied/#page-variable) variable was `undefined`. 

example.njk
```nunjucks
{% macro example %}
<p>{{ 'some.thing' | i18n }}</p>

<p>{{ 'page: '+page.lang }}</p>
{% endmacro %}
```

output.html
```html
<p>Esto es un fallback</p>

<p>page: undefined</p>
```

Inside layouts it works fine, but I wonder if it is posible to acces to the `page` variable from a macro or I'm doing something wrong.

**Comment by pdehaan** [comment]
You might have to explicitly pass the `page` object to your macro.

```njk
<!-- src/index.njk -->
{%- import "macros.njk" as macros -%}

<article>
  {{ macros.field("name", "", "text", page) }}
</article>
```

```njk
<!-- src/_includes/macros.njk -->
{% macro field(name, value="", type="text", page={}) %}
  <div class="field">
    <input type="{{ type }}" name="{{ name }}"
          value="{{ value | escape }}" />
    <!-- {{ page | dump(2) | safe }} -->
  </div>
{% endmacro %}

```

### OUTPUT
```html
<article>
  <div class="field">
    <input type="text" name="name" value="" />
    <!-- {
      "date": "2023-03-08T18:25:38.768Z",
      "inputPath": "./src/index.njk",
      "fileSlug": "",
      "filePathStem": "/index",
      "outputFileExtension": "html",
      "templateSyntax": "njk",
      "url": "/",
      "outputPath": "www/index.html"
    } -->
  </div>
</article>
```

**Comment by TheMen4ce** [comment]
Passing the `page` variable to every macro seems a bit hacky/verbose to me.

What's the reason for `page` not being available? Especially for i18n, the page object is very helpful. E.g. currently it's not possible to use the `locale_url` filter in a macro.
